release-tool.ps1

```diff
@@ -0,0 +1,456 @@
+    }
+}
+
+function Test-WorkingTreeClean {
+    $(git diff-index --quiet HEAD --)
+    if ($LASTEXITCODE) {
+        throw "Current working tree is not clean! Please commit or unstage any changes."
+    }
+}
+
+function Invoke-VSToolchain([String] $Toolchain, [String] $Path, [String] $Arch) {
+    # Find Visual Studio installations
+    $vs = Get-CimInstance MSFT_VSInstance
+    if ($vs.count -eq 0) {
+        $err = "No Visual Studio installations found, download one from https://visualstudio.com/downloads."
+        $err = "$err`nIf Visual Studio is installed, you may need to repair the install then restart."
+        throw $err
+    }
+
+    $VSBaseDir = $vs[0].InstallLocation
+    if ($Toolchain) {
+        # Try to find the specified toolchain by name
+        foreach ($_ in $vs) {
+            if ($_.Name -eq $Toolchain) {
+                $VSBaseDir = $_.InstallLocation
+                break
+            }
+        }
+    }
+    elseif ($vs.count -gt 1) {
+        # Ask the user which install to use
+        $i = 0
+        foreach ($_ in $vs) {
+            $i = $i + 1
+            $i.ToString() + ") " + $_.Name | Write-Host
+        }
+        $i = Read-Host -Prompt "Which Visual Studio installation do you want to use?"
+        $i = [Convert]::ToInt32($i, 10) - 1
+        if ($i -lt 0 -or $i -ge $vs.count) {
+            throw "Invalid selection made"
+        }
+        $VSBaseDir = $vs[$i].InstallLocation
+    }
+    
+    # Bootstrap the specified VS Toolchain
+    Import-Module "$VSBaseDir\Common7\Tools\Microsoft.VisualStudio.DevShell.dll"
+    Enter-VsDevShell -VsInstallPath $VSBaseDir -Arch $Arch -StartInPath $Path | Write-Host
+    Write-Host # Newline after command output
+}
+
+function Invoke-Cmd([string] $command, [string[]] $options = @(), [switch] $maskargs, [switch] $quiet) {
+    $call = ('{0} {1}' -f $command, ($options -Join ' '))
+    if ($maskargs) {
+        Write-Host "$command <masked>" -ForegroundColor DarkGray
+    }
+    else {
+        Write-Host $call -ForegroundColor DarkGray
+    }
+    if ($quiet) {
+        Invoke-Expression $call > $null
+    } else {
+        Invoke-Expression $call
+    }
+    if ($LASTEXITCODE -ne 0) {
+        throw "Failed to run command: {0}" -f $command
+    }
+    Write-Host #insert newline after command output
+}
+
+function Invoke-SignFiles([string[]] $files, [string] $key, [string] $time) {
+    if (-not (Test-Path -Path "$key" -PathType leaf)) {
+        throw "Appsign key file was not found! ($key)"
+    }
+    if ($files.Length -eq 0) {
+        return
+    }
+
+    Write-Host "Signing files using $key"  -ForegroundColor Cyan
+    $KeyPassword = Read-Host "Key password: " -MaskInput
+
+    foreach ($_ in $files) {
+        Write-Host "Signing file '$_' using Microsoft signtool..."
+        Invoke-Cmd "signtool" "sign -f `"$key`" -p `"$KeyPassword`" -d `"KeePassXC`" -td sha256 -fd sha256 -tr `"$time`" `"$_`"" -maskargs
```

Comment:
backtick is the escape character for powershell, not the exec function like in bash. so `" is escaping the quote character.

Reply:
I know, but I meant that the parameters should be passed individually as a list, like you would do normally when invoking external commands instead of pre-assembling a string that is evaluated as a shell. But that would require us to use something other than `Invoke-Cmd` (e.g., `Invoke-Command` or so, not sure what the best thing would be). Doing it like this could potentially lead to problems with disallowed characters in one of the variables (and it's less efficient).